baremetal/metal3data_manager_test.go

```diff
@@ -1307,13 +1288,11 @@ var _ = Describe("Metal3Data manager", func() {
 	)
 
 	type testCaseReleaseAddressFromPool struct {
```

Comment:
now that we dont have expectAddresses and ReleaseAddressFromPool is only returning errors, these testcases are not making so much sense because we are not checking any negative testcases. I would expect a testcase here where expectError is true and probably we should mention the false cases also explicitly.

Reply:
Testing negative cases requires to inject errors into the api client, which fakeclient does not support. I think I've wrapped it to do so in the past, let me see if I can find that.

Reply:
Maybe using mocks? Like here in the [metal3machine_controller_test.go](https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/controllers/metal3machine_controller_test.go#L94)

Reply:
I've cleaned up the tests now and added one that injects an error on deletion.